--- a/src/core/qmlglobal.cpp
+++ b/src/core/qmlglobal.cpp
@@ -74,9 +74,9 @@ QuickshellTracked::QuickshellTracked() {

 	if (guiApp != nullptr) {
 		// clang-format off
-		QObject::connect(guiApp, &QGuiApplication::primaryScreenChanged, this, &QuickshellTracked::updateScreens);
-		QObject::connect(guiApp, &QGuiApplication::screenAdded, this, &QuickshellTracked::updateScreens);
-		QObject::connect(guiApp, &QGuiApplication::screenRemoved, this, &QuickshellTracked::updateScreens);
+		QObject::connect(guiApp, &QGuiApplication::primaryScreenChanged, this, &QuickshellTracked::updateScreens, Qt::QueuedConnection);
+		QObject::connect(guiApp, &QGuiApplication::screenAdded, this, &QuickshellTracked::updateScreens, Qt::QueuedConnection);
+		QObject::connect(guiApp, &QGuiApplication::screenRemoved, this, &QuickshellTracked::updateScreens, Qt::QueuedConnection);
 		// clang-format on

 		this->updateScreens();
